### PR TITLE
Adding Notify to Hipchat output plugin.

### PR DESCRIPTION
**Pull request by Grejeru**
Based on pushbullet plugin I've wrote hipchat plugin to notify Hipchat room with notifications.
Inside plugin code there are exact informations how to configure it, if it'll be merged I'll write proper docs on Trac too.

### CODE REVIEW

File: flexget/plugins/output/hipchat.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        auth_token: <AUTH_TOKEN>
+        room_key: <ROOM_KEY>
+        [color: green|yellow|red|grey|purple (default: green)]
+        [notify: true|false (default: false)]
+        [title: <TITLE> (default: "New download started:")]
+        [message: <MESSAGE_BODY> (default: "{{ series_name }} {{ series_id }} {{ quality }}" -- accepts Jinja2)]
+        [url: <HIPCHAT_URL> (default: https://api.hipchat.com)]
+    Configuration parameters are also supported from entries (eg. through set).
+    """
+    default_message = ('{{ series_name }} {{ series_id }} {{ quality }}')
```
Comment (paranoidi) [inline flexget/plugins/output/hipchat.py]: Default should not rely on other plugins. Use title/url.

**Comment by paranoidi** [comment]
Looks quite good
+1

**Comment by Grejeru** [comment]
@paranoidi I've corrected default_message as You've asked.
Still I don't know how to add hipchat to logger, as currently my plugin works when I'm having:
```
log = logging.getLogger('pushbullet')
```

**Comment by Grejeru** [comment]
@paranoidi thx, it's now corrected.
All problems resolved :-)

File: flexget/plugins/output/hipchat.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        'required': ['auth_token', 'room_key'],
+        'additionalProperties': False
+    }
+
+    # Run last to make sure other outputs are successful before sending notification
+    @plugin.priority(0)
+    def on_task_output(self, task, config):
+
+        for entry in task.accepted:
+            auth_token = config['auth_token']
```
Comment (liiight) [inline flexget/plugins/output/hipchat.py]: `auth_token` can be a list as well, so you need to iterate over it I believe. Same for `room_key`.

Comment (Grejeru) [inline flexget/plugins/output/hipchat.py]: @liiight yes, but it's always paired, auth_token + room_key and I don't know how it can be forced to be paired.

Comment (liiight) [inline flexget/plugins/output/hipchat.py]: if there's a one to one relationship between `auth_key` and `room_key` you can set at as it's own object:
```python
auth = {
        'type': 'object',
        'properties': {
            'auth_key': {'type': 'string'},
            'room_key': {'type': 'string'}
        },
        'required': ['auth_key', 'room_key'],
        'additionalProperties': False
    }
```
And then use:
```python
schema = {
        'type': 'object',
        'properties': {
            'auth': one_or_more(auth),
            'color': {'type': 'string', 'default': 'green'},
            'notify': {'type': 'string', 'default': 'false'},
            'title': {'type': 'string', 'default': 'New download started:'},
            'message': {'type': 'string', 'default': default_message},
            'url': {'type': 'string', 'default': 'https://api.hipchat.com'},
        },
        'required': ['auth'],
        'additionalProperties': False
    }
```
Again, this is valid ONLY if there's a 1-to-1 relationship between auth_key and room_key but you get the idea. Then just iterate over the config and verify that you always use a list:
```python
if isinstance(config.get('auth'), basestring):
    config['auth'] = [config['auth']]

**Comment by gazpachoking** [comment]
@Grejeru @liiight So, are we waiting on some additional config format changes, or is this good to go?

**Comment by liiight** [comment]
Still waiting

**Comment by liiight** [comment]
@grejeru don't you wanna change this so we can merge?

**Comment by Grejeru** [comment]
Sorry, had lot on my head, I'll try to update it till EoW.

**Comment by liiight** [comment]
Reopen if relevant

**Pull request closed without merging**